sale_order_product_recommendation/wizards/sale_order_recommendation.py

```diff
@@ -80,42 +80,37 @@ def _generate_recommendations(self):
         )
         found_dict = {l["product_id"][0]: l for l in found_lines}
         RecomendationLine = self.env["sale.order.recommendation.line"]
+        wiz_line_ids = self.env["sale.order.recommendation.line"]
         existing_product_ids = []
         # Add products from sale order lines
         for line in self.order_id.order_line:
             found_line = found_dict[line.product_id.id]
-            new_line = RecomendationLine.new({
+            new_line = RecomendationLine.create({
                 "product_id": line.product_id.id,
                 "times_delivered": found_line["product_id_count"],
                 "units_delivered": found_line["qty_delivered"],
                 "units_included": line.product_uom_qty,
                 "sale_line_id": line.id,
             })
-            self.line_ids += new_line
+            wiz_line_ids |= new_line
             existing_product_ids.append(line.product_id.id)
         # Add those recommendations too
         i = 0
         for line in found_lines:
             if line["product_id"][0] in existing_product_ids:
                 continue
-            new_line = RecomendationLine.new({
+            new_line = RecomendationLine.create({
                 "product_id": line["product_id"][0],
                 "times_delivered": line["product_id_count"],
                 "units_delivered": line["qty_delivered"],
             })
-            self.line_ids += new_line
+            wiz_line_ids |= new_line
             # limit number of results. It has to be done here, as we need to
             # populate all results first, for being able to select best matches
             i += 1
             if i == self.line_amount:
                 break
-
-    @api.model
-    def create(self, vals):
-        if 'line_ids' in vals:
-            vals['line_ids'] = [
-                line for line in vals['line_ids'] if line[2]["is_modified"]]
-        return super(SaleOrderRecommendation, self).create(vals)
+        self.line_ids = wiz_line_ids
```

Comment:
Or is it possible that all those `+=` mess things up? Have you tried just storing all new records in a separate variable and assigning it just once at the end like here?

BTW you can remove line 56 with this approach.